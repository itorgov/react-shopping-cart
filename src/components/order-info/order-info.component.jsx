import React from 'react';
import {useSelector, useDispatch} from "react-redux";

import {Input} from "../input/input.component";
import {updateNote} from '../../redux/cart/cart.actions';

import styles from './order-info.module.scss';

export const OrderInfo = () => {
    const note = useSelector(state => state.cart.note);
    const dispatch = useDispatch();

    return (
        <div className={styles.info}>
            <Input
                id='note'
                name='note'
                type='text'
                placeholder='Some words to our team'
                label='Add a note'
                value={note}
                onChange={e => dispatch(updateNote(e.target.value))}
            />
        </div>
    )
};
