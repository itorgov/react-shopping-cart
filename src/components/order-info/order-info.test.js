import React from "react";
import {mount} from "enzyme";
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store'

import {OrderInfo} from './order-info.component';

let wrapper;
let store;

beforeEach(() => {
    const mockStore = configureStore();
    const initialState = {
        cart: {
            items: [
                {
                    id: 1,
                    name: "Test name",
                    price: 3900,
                    quantity: 2,
                }
            ],
            note: 'Test note value',
        }
    };
    store = mockStore(initialState);
    wrapper = mount(<Provider store={store}><OrderInfo /></Provider>);
});

describe('<OrderInfo />', () => {
    it('expect to render', () => {
        expect(wrapper.find('[name="note"]').prop('value')).toEqual('Test note value');
        expect(wrapper).toMatchSnapshot();
    });
});
