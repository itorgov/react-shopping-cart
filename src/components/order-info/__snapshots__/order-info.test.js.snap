// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderInfo /> expect to render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <OrderInfo>
    <div
      className="info"
    >
      <Input
        id="note"
        label="Add a note"
        name="note"
        onChange={[Function]}
        placeholder="Some words to our team"
        type="text"
        value="Test note value"
      >
        <div
          className="group"
        >
          <label
            className="label"
            htmlFor="note"
          >
            Add a note
          </label>
          <input
            className="input"
            id="note"
            onChange={[Function]}
            placeholder="Some words to our team"
            type="text"
            value="Test note value"
          />
        </div>
      </Input>
    </div>
  </OrderInfo>
</Provider>
`;
